src/components/ReadWrite.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { FirestoreCollection, withFirestore } from 'react-firestore';
+
+function ReadWrite(props) {
+  const { firestore } = props;
+  const [enteredValue, setEnteredValue] = useState();
+
+  const addMsg = e => {
+    e.preventDefault();
+    firestore.collection('example').add({
+      msg: enteredValue,
+    });
+  };
+
+  return (
+    <div>
```

Comment:
Just a suggestion, not a blocker: Since this `div` isn't doing anything special, you can use a `React.Fragment` instead. This will help you clean up the code a bit by not adding extra nodes to the DOM. Check it out: https://reactjs.org/docs/fragments.html